editor/import/resource_importer_texture_atlas.cpp

```diff
@@ -306,10 +306,8 @@ Error ResourceImporterTextureAtlas::import_group_file(const String &p_group_file
 
 	//update cache if existing, else create
 	Ref<Texture2D> cache;
-	if (ResourceCache::has(p_group_file)) {
-		Resource *resptr = ResourceCache::get(p_group_file);
-		cache.reference_ptr(resptr);
-	} else {
+	cache = ResourceCache::get_ref(p_group_file);
+	if (!cache.is_valid()) {
```

Comment:
Would it be interesting that `ResourceCache` had a function like `get_or_create_ref()` one can use in cases like this to ensure no race conditions between two threads trying to create a resource with the same path?

Reply:
I think with this PR there should be no risk or race conditions any longer.

Reply:
It could, but that would be quite a wierd use case, in which case its fine. Normally you load things from the ResourceLoader, which checks in a central place for these things.

Reply:
Fair enough!